pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -230,6 +234,20 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		if err != nil {
 			return err
 		}
+
+		runCommand := pushDevfileCommands[devfilev1.RunCommandGroupKind]
+		if parameters.Debug {
+			runCommand = pushDevfileCommands[devfilev1.DebugCommandGroupKind]
+		}
+
+		// wait for a second
+		wait := time.After(supervisorDStatusWaitTimeInterval * time.Second)
```

Comment:
why wait here?

Reply:
It's a requirement mentioned in https://github.com/openshift/odo/issues/4197#issuecomment-724113853